third_party/nginx_latency_status_module/src/ngx_http_latency_stub_status_module.c

```diff
@@ -0,0 +1,306 @@
+  return *num;
+}
+
+
+// Parse the config for the store_latency directive.
+// It expects 3 arguments, base, scaled_factor, and max_exponent,  defining the buckets
+// for the latency distributions that will be stored.
+// There will be max_exponent + 2 buckets with the boundaries
+// [0, scaled_factor * base ^ i) for i = 0
+// [scaled_factor * base ^ (i - 1), scaled_factor * base ^ i ) for 0 < i <= max_exponent
+// [scaled_factor * base ^ (i - 1), infinity) for i = max_exponent + 1
+char* ngx_http_latency(ngx_conf_t* cf, ngx_command_t* cmd, void* conf){
+  ngx_http_latency_main_conf_t* main_conf;
+  ngx_int_t base, scale_factor, max_exponent;
+  ngx_int_t parse_result;
+  ngx_str_t *config_value;
+
+  config_value = cf->args->elts;
+  main_conf = (ngx_http_latency_main_conf_t*) conf;
+
+  parse_result = ngx_parse_int(cf, 1, &base);
+  if (parse_result == NGX_ERROR) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: base \"%V\"", &config_value[1]);
+    return NGX_CONF_ERROR;
+  }
+
+  parse_result = ngx_parse_int(cf, 2, &scale_factor);
+  if (parse_result == NGX_ERROR) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: scaled_factor \"%V\"", &config_value[2]);
+    return NGX_CONF_ERROR;
+  }
+
+  parse_result = ngx_parse_int(cf, 3, &max_exponent);
+  if (parse_result == NGX_ERROR) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: max_value \"%V\"", &config_value[3]);
+    return NGX_CONF_ERROR;
+  }
+
+  if (base <= 1) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: base %d. It must be greater than 1", base);
+    return NGX_CONF_ERROR;
+  }
+  if (scale_factor <= 0) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_sub: scaled_factor %d. It must be greater than 0", scale_factor);
+    return NGX_CONF_ERROR;
+  }
+  if (max_exponent < 1) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_sub: max_exponent %d. It must be at least 1", max_exponent);
+    return NGX_CONF_ERROR;
+  }
+
+  main_conf->base = base;
+  main_conf->scale_factor = scale_factor;
+  main_conf->max_exponent = max_exponent;
+  shm_max_exponent = max_exponent;
+
+  ngx_int_t* bucket_bounds;
+  ngx_int_t power = 1;
+  bucket_bounds = ngx_palloc(cf->pool, sizeof(ngx_int_t) * (max_exponent + 1));
+  if (bucket_bounds == NULL) {
+    return NGX_CONF_ERROR;
+  }
+  for (ngx_int_t i = 0; i <= max_exponent; i++) {
+    bucket_bounds[i] = scale_factor * power;
+    power *= base;
+  }
+  main_conf->latency_bucket_bounds = bucket_bounds;
+
+  ngx_shm_zone_t *shm_zone;
+  ngx_str_t *shm_name;
+
+  shm_name = ngx_palloc(cf->pool, sizeof *shm_name);
+  shm_name->len = sizeof("latency_shared_memory") - 1;
```

Comment:
Should `len` be `sizeof("shared_memory") - 1`, or the other way around for `data`?

Reply:
I've set data to match this.